server/main.lua

```diff
@@ -94,5 +82,91 @@ lib.callback.register('qbx_truckerjob:server:spawnVehicle', function(source, mod
 end)
 
 RegisterNetEvent("QBCore:Server:OnPlayerUnload", function (client)
-    currentTruckers[client] = nil
-end)
\ No newline at end of file
+    locations[client] = nil
+end)
+
+local function isNotLocationDone(location, current)
+    for i=1,#location.done do
+        if location.done[i] == current then
+            return false
+        end
+    end
+
+    return true
+end
+
+local function getPlayer(source)
+    local player = exports.qbx_core:GetPlayer(source)
+    if not player or player.PlayerData.job.name ~= "trucker" then return end
+    return player
+end
+
+local function getPlayerRoutes(source)
+    local player = getPlayer(source)
+    if not player then return end
+    local routes = 0
+    local location = locations[player.PlayerData.source]
+    if location then
+        routes = location.done
+    end
+
+    return player, routes
+end
+
+local function getReward(player)
+    local chance = math.random(1, 100)
+    if chance > 26 then return end
+
+    player.Functions.AddItem("cryptostick", 1, false)
+end
+
+lib.callback.register('qbx_truckerjob:server:doneJob', function (source)
```

Comment:
This naming seems odd. A callback is a function and so should be named with a leading verb. Consider renaming doneJob to something like getNewLocation

Reply:
I changed it to `getNewTask`.